WordPress/Classes/ViewRelated/Site Creation/Site Name/SiteNameView.swift

```diff
@@ -0,0 +1,199 @@
+
+import UIKit
+import WordPressShared
+
+/// content view for SiteNameViewController
+class SiteNameView: UIView {
+
+    private let siteName: String
+
+    // MARK: UI
+
+    /// Title
+    private lazy var titleLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.font = WPStyleGuide.serifFontForTextStyle(.largeTitle, fontWeight: .semibold)
+        label.adjustsFontForContentSizeCategory = true
+        label.numberOfLines = Metrics.numberOfLinesInTitle
+        label.textAlignment = .center
+        return label
+    }()
+
+    // used to add left and right padding to the title
+    private lazy var titleLabelView: UIView = {
+        let view = UIView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(titleLabel)
+        return view
+    }()
+
+    /// Subtitle
+    private lazy var subtitleLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.font = WPStyleGuide.fontForTextStyle(.body)
+        label.adjustsFontForContentSizeCategory = true
+        label.textColor = .secondaryLabel
+        label.setText(TextContent.subtitle)
+        label.numberOfLines = Metrics.numberOfLinesInSubtitle
+        label.textAlignment = .center
+        return label
+    }()
+
+    // used to add left and right padding to the subtitle
+    private lazy var subtitleLabelView: UIView = {
+        let view = UIView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(subtitleLabel)
+        return view
+    }()
+
+    /// Search bar
+    private lazy var searchBar: UISearchBar = {
+        let searchBar = UISearchBar()
```

Comment:
Curious - what's the benefit for this view to use a `UISearchBar` over a `UITextField`? I do see we're using the delegate method for the button. Were there plans to make this more like a search in the future?

Reply:
The main goal here is to use newer standard components. Looking at the documentation, using the new `UISearchTextField` ([ref](https://developer.apple.com/documentation/uikit/uisearchtextfield)) seems suitable for a more custom usage, , while in our case `UISearchBar` ([ref](https://developer.apple.com/documentation/uikit/uisearchbar)) does it, and it seems convenient.

Reply:
> The placeholder text is showing "Search" so perhaps we can update it to something more appropriate for this view?

This will be the name of the vertical, but since this is just UI, it will be fixed in another PR. I've added it to the known issues

Reply:
Thanks @Gio2018 for the info!